File: kata-others/kata-python/the_sum_of_things.py
import fileinput


def main():
    for line in fileinput.input():
        parts = line.split()
        result = 0
        for p in parts:
            try:
                result += int(p, 0)
            except:
                result += ord(p)
        print(result)

if __name__ == '__main__':
    main()

File: kata-others/kata-php/the_sum_of_things.php
<?php

$f = fopen( 'php://stdin', 'r' );

while( $line = fgets( $f ) ) {
    $line = trim($line);
    $split = explode(" ", $line);
    $result = 0;
    foreach ($split as $value) {
        if (is_numeric($value)) {
            $result += $value;
        } else if (strlen($value) == 1) {
            $result += ord($value);
        } else if ($value[1] === "b") {
            $result += bindec($value);
        } else if ($value[1] === "o") {
            $result += octdec($value);
        } else if ($value[1] === "x") {
            $result += hexdec($value);
        }
    }
    echo $result . "\n";
}   

fclose( $f );

?>
